portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -0,0 +1,46 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.KeyRange;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/delete-data")
+public class DeleteServlet extends HttpServlet {
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+     Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+     PreparedQuery results = datastore.prepare(query);
+
+     ArrayList<Key> keys = new ArrayList<Key>();
+
+     for (Entity entity : results.asIterable()) {
+         Key key = entity.getKey();
+         keys.add(key);
+     }
+     for(Key key: keys){
+         datastore.delete(key);
+     }
```

Comment:
this logic looks like it could be combined. Instead of looping over each result, storing the keys, then looping over the keys, you should be able to call datastore.delete(entity.getKey()) in the for loop.